### PR TITLE
basic deploy tool using climb

### PR DESCRIPTION
**Pull request by dakom**

### CODE REVIEW

**Review by ethanfrey — APPROVED** [review]
First of all, I fixed dev-net and I tested your deploy script there and it worked perfectly.
I hadn't updated for the last 3 weeks of gRPC server changes.

Seconds, bar a number of minor comments, this looks quite good as a start. I'd be happy for you to merge this once you resolve that little cleanup (but not close the issue), then can start new PRs to improve if needed.

eg. we want to allow customizable operator addresses, required percentage, timeouts etc with reasonable defaults. But that is minor to gettting this whole thing working

File: tools/deploy/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+name = "deploy-cli"
+version = "0.0.0"
+edition = "2021"
+license = "Apache-2.0"
+
+[dependencies]
+### Remote dependencies ###
+layer-climb = { git = "https://github.com/Lay3rLabs/climb.git", rev = "bf2ce73be2b46957c22885fda33466ec6b10e087"}
+### Local dependencies ###
+lavs-mock-operators = {path = "../../contracts/mock-operators"}
```
[the remark this answers is not in the document]
Comment (dakom) [reply tools/deploy/Cargo.toml]: done in https://github.com/Lay3rLabs/avs-toolkit/pull/21/commits/ae65704a7c5bf1f6850719e308826301cb274bc2

File: tools/deploy/config.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+    "chains": {
+        "local": {
+            "chain_id": "slay3r-local",
+            "rpc_endpoint": "http://localhost:26657",
+            "grpc_endpoint": "http://localhost:9090",
+            "grpc_web_endpoint": "http://localhost:9091",
```
[the remark this answers is not in the document]
Comment (dakom) [reply tools/deploy/config.json:7]: yup, this is just a weird mistake, I opened an issue at https://github.com/Lay3rLabs/climb/issues/4

File: tools/deploy/config.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            "gas_amount": "0.025",
+            "gas_denom": "uslay",
+            "address_kind": {"cosmos":{"prefix":"slay3r"}}
+        },
+        "testnet": {
+            "chain_id": "slay3r-dev",
+            "rpc_endpoint": "https://lcd.dev-cav3.net",
+            "grpc_endpoint": "https://grpc.dev-cav3.net",
+            "grpc_web_endpoint": "https://grpc.dev-cav3.net",
```
[the remark this answers is not in the document]
Comment (dakom) [reply tools/deploy/config.json]: hmmm... yeah, I updated the port but still getting errors. changed the config here in https://github.com/Lay3rLabs/avs-toolkit/pull/21/commits/9bfc955eb30b24836277806f6ac37464fe4f61d7 and  opened https://github.com/Lay3rLabs/layer-sdk/issues/161

File: tools/deploy/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+                );
+            }
+            if !verifier_simple_path.exists() {
+                bail!(
+                    "Verifier Simple contract not found at {} (try running collect_wasm.sh)",
+                    verifier_simple_path.display()
+                );
+            }
+
+            // TODO - fix account sequence stuff in climb so we can do this all concurrently
```
[the remark this answers is not in the document]
Comment (dakom) [reply tools/deploy/src/main.rs:60]: I have a solution for this :) will open a separate PR after this which does it, but ofc you can decide to reject it and we can delete the TODO - just don't want that to block here, so resolving the comment but _not_ forgetting about it

File: Cargo.toml [hunk 5]
```diff
@@ -39,6 +39,15 @@ derivative    = "2"
 test-case     = "3.3.1"
 clap = { version = "4.5.1", features = ["derive", "env"] }
 
+# tooling deps
+tokio = { version = "1", features = ["full"] }
```
[the remark this answers is not in the document]
Comment (dakom) [reply Cargo.toml:43]: done in https://github.com/Lay3rLabs/avs-toolkit/pull/21/commits/ae65704a7c5bf1f6850719e308826301cb274bc2

yeah, 0.1.0 would be great, feels pretty close :)

File: tools/deploy/README.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+`cargo run -- --help` to see all the commands
```
[the remark this answers is not in the document]
Comment (dakom) [reply tools/deploy/README.md:3]: done in https://github.com/Lay3rLabs/avs-toolkit/pull/21/commits/ee7be56020db19bd1e62eefee0300f57cd70de4e

**Pull request merged**